Plugins/Detray/src/DetrayConverter.cpp

```diff
@@ -0,0 +1,318 @@
+
+  auto [surfaceAdjusted, insidePointer] = orientedSurfaces[ip];
+
+  // Single link detected - just write it out, we use the oriented surface
+  // in order to make sure the size is adjusted
+  if (singleLink != nullptr) {
+    // Single link can be written out
+    std::size_t vLink = findVolume(singleLink->object(), detectorVolumes);
+    auto portalPayload = convertSurface(gctx, *surfaceAdjusted, true);
+    portalPayload.mask.volume_link.link = vLink;
+    portals.push_back(portalPayload);
+  } else {
+    // Multi link detected - 1D
+    auto multiLink1D =
+        dynamic_cast<const Experimental::BoundVolumesGrid1Navigation*>(
+            instance);
+    if (multiLink1D != nullptr) {
+      // Resolve the multi link 1D
+      auto boundaries =
+          multiLink1D->indexedUpdater.grid.axes()[0u]->getBinEdges();
+      const auto& cast = multiLink1D->indexedUpdater.casts[0u];
+      const auto& transform = multiLink1D->indexedUpdater.transform;
+      const auto& volumes = multiLink1D->indexedUpdater.extractor.dVolumes;
+
+      // Apply the correction from local to global boundaries
+      ActsScalar gCorr = VectorHelpers::cast(transform.translation(), cast);
+      std::for_each(boundaries.begin(), boundaries.end(),
+                    [&gCorr](ActsScalar& b) { b -= gCorr; });
+
+      // Get the volume indices
+      auto surfaceType = surfaceAdjusted->type();
+      std::vector<unsigned int> vIndices = {};
+      for (const auto& v : volumes) {
+        vIndices.push_back(findVolume(v, detectorVolumes));
+      }
+
+      // Pick the surface dimension - via poly
+      std::array<ActsScalar, 2u> clipRange = {0., 0.};
+      std::vector<ActsScalar> boundValues = surfaceAdjusted->bounds().values();
+      if (surfaceType == Surface::SurfaceType::Cylinder && cast == binZ) {
+        ActsScalar zPosition = surfaceAdjusted->center(gctx).z();
+        clipRange = {
+            zPosition - boundValues[CylinderBounds::BoundValues::eHalfLengthZ],
+            zPosition + boundValues[CylinderBounds::BoundValues::eHalfLengthZ]};
+      } else if (surfaceType == Surface::SurfaceType::Disc && cast == binR) {
+        clipRange = {boundValues[RadialBounds::BoundValues::eMinR],
+                     boundValues[RadialBounds::BoundValues::eMaxR]};
+      } else {
+        throw std::runtime_error(
+            "PortalDetrayConverter: surface type not (yet) supported for "
+            "detray "
+            "conversion, only cylinder and disc are currently supported.");
+      }
+
+      // Need to clip the parameter space to the surface dimension
+      std::vector<unsigned int> clippedIndices = {};
+      std::vector<ActsScalar> clippedBoundaries = {};
+      clippedBoundaries.push_back(clipRange[0u]);
+      for (const auto [ib, b] : enumerate(boundaries)) {
+        if (ib > 0) {
+          unsigned int vI = vIndices[ib - 1u];
+          ActsScalar highEdge = boundaries[ib];
+          if (boundaries[ib - 1] >= clipRange[1u]) {
```

Comment:
This is an example of what I'm talking about in the general comment.

Reply:
I see, I agree that we should write that to explicit retrieval, I will put it onto my list.